shared/lib_battery_voltage.cpp

```diff
@@ -379,37 +405,49 @@ double voltage_dynamic_t::calculate_max_charge_w(double q, double qmax, double,
 
 using namespace std::placeholders;
 
-double voltage_dynamic_t::calculate_max_discharge_w(double q, double qmax, double, double *max_current) {
+double voltage_dynamic_t::calculate_max_discharge_w(double q, double qmax, double kelvin, double *max_current) {
     q /= params->num_strings;
     qmax /= params->num_strings;
 
-    double current = 0., vol = 0;
+    double current = 0.;
+    double vol = params->dynamic.Vcut + 0.012;
+    //double vol = 0;
     double incr = q / 10;
-    double max_p = 0, max_I = 0;
-    while (current * params->dt_hr < q - tolerance && vol >= 0) {
+    //double vol_diff = params->dynamic.Vcut - vol;
+    double max_p = 0, max_I = 0, max_V = 0;
+    double C, x;
+    while (current * params->dt_hr < q - tolerance && vol >= params->dynamic.Vcut + 0.012 ) {
         vol = voltage_model_tremblay_hybrid(qmax, current, q - current * params->dt_hr);
         double p = current * vol;
         if (p > max_p) {
             max_p = p;
             max_I = current;
+            max_V = vol;
+            if (params->dynamic.Vcut != 0) {
+                C = (-1 * params->dynamic.Vcut + _E0 - params->resistance * q - current * params->dt_hr + _A * exp(-_B0 * qmax)) / _K;
+                x = qmax / (C - 1);
+                params->dynamic.Qfull_mod = x;
+                 
+            }
```

Comment:
Isn't this now moved into `voltage_model_tremblay_hybrid`?

This function should not make any changes to `params` or `state`

Reply:
Removed.